cli/command/secret/create.go

```diff
@@ -46,19 +50,26 @@ func runSecretCreate(dockerCli command.Cli, options createOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	var in io.Reader = dockerCli.In()
-	if options.file != "-" {
-		file, err := system.OpenSequential(options.file)
-		if err != nil {
-			return err
-		}
-		in = file
-		defer file.Close()
+	if options.driver != "" && options.file != "" {
+		return errors.Errorf("When using secret driver secret data must be empty")
 	}
 
-	secretData, err := ioutil.ReadAll(in)
-	if err != nil {
-		return errors.Errorf("Error reading content from %q: %v", options.file, err)
+	var secretData []byte
+	if options.driver == "" {
+		var in io.Reader = dockerCli.In()
+		if options.file != "-" {
+			file, err := system.OpenSequential(options.file)
+			if err != nil {
+				return err
+			}
+			in = file
+			defer file.Close()
+		}
+		var err error
+		secretData, err = ioutil.ReadAll(in)
+		if err != nil {
+			return errors.Errorf("Error reading content from %q: %v", options.file, err)
+		}
```

Comment:
I know you've just adding another condition here, but now that this function is growing I think it would be appropriate to extract this as a new function.

```go
secretData, err := readSecretData(options.file, dockerCli.In())
...
```

Even the initial `options.file != ""` check could be inside the function.